src/core/CurveTricrypto.sol

```diff
@@ -35,6 +35,38 @@ interface ICurveTricryptoCallback {
     ) external;
 }
 
+library FastCurveTricrypto {
+    function fastEncodeExchangeExtended(
+        uint256 sellIndex,
+        uint256 buyIndex,
+        uint256 sellAmount,
+        uint256 minBuyAmount,
+        address receiver
+    ) internal pure returns (bytes memory data) {
+        assembly ("memory-safe") {
+            data := mload(0x40)
+            
+            // callback is stored at 
+            // mstore(add(0x104, data), shl(0xe0, 0x6370a85c))
+            // which is analogous to the next line
+            // except that anything from add(0x108, data) on
+            // is dirty, but that is not a problem as only the
+            // initial 4 bytes at add(0x104, data) are used
+            mstore(add(0xe8, data), 0x6370a85c) // selector for `curveTricryptoSwapCallback(address,address,address,uint256,uint256)`
+            mstore(add(0xe4, data), receiver)
+            codecopy(add(0xa4, data), codesize(), 0x40) // useEth and payer (both zeroed)
```

Comment:
`receiver` needs to be masked of dirty bits. This can be achieved by adjusting the range of bytes zeroized by `codecopy`

```suggestion
            mstore(add(0xe4, data), receiver)
            codecopy(add(0xa4, data), codesize(), 0x4c) // useEth and payer (both zeroed); clear dirty bits in `receiver`
```